### PR TITLE
Adapt to xscen-509

### LINKED ISSUE #13: attributes created by intake-esm contain illegal characters
`intake-esm` creates global attributes based on the catalog. By default, they follow the format: `intake_esm_attr/(x)`. 

However, `/` is an illegal character when, in xarray, `engine="netcdf4"`. We have no control over that character, and so this will need a fix in `intake_esm` itself. In the meantime, a fix is needed in `xscen` too.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] `pre-commit` hooks are installed/active in my local clone (`$ pre-commit install`)
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #13 
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] If a merge request has been made in parallel to this PR in xscen-notebooks, it is merged and the submodules have been updated. [MR#6](https://gitlab.ouranos.ca/scenario/xscen-notebooks/merge_requests/6)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Use `cat:` to prefix dataset attributes generated from catalogs.
* New `xscen.utils.get_cat_attrs` that outputs a dictionary of those attributes without the prefix.

### Does this PR introduce a breaking change?
Yes (but it comes from `intake-esm`)

### Other information:
The intake-esm change also affects how lists are parsed : 
https://github.com/intake/intake-esm/blob/ba4bb59ca1174369ff45a09aed030db0a7aa49d2/intake_esm/source.py#L100-L105

I.e.: every object that is iterable (like lists or tuples) is formatted as a comma-separated string. I think this doesn't break xscen, but might be worth noting. The only list-like attribute we usually use is `variable`.

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,14 @@
 History
 =======
 
+v0.3.2 (2022-08-23)
```
Comment (RondeauG) [inline HISTORY.rst:5]: This is because it will get its own release, right ?

File: xscen/utils.py [hunk 2]
```diff
@@ -369,13 +378,13 @@ def clean_up(
...
         - starting with a '^' means check if the string starts with the substring.
-        eg. {'global': ['necessary note', '^cat/'], 'tasmax': 'new_name'}
+        eg. {'global': ['necessary note', '^cat:'], 'tasmax': 'new_name'}
     add_attrs: dict
         Dictionary where the keys are the variables and the values are a another dictionary of attributes.
         For global attrs, use the key 'global'.
         eg. {'global': {'title': 'amazing new dataset'}, 'tasmax': {'note': 'important info about tasmax'}}
     change_attr_prefix: str
-        Replace "cat/" in the catalogue global attrs by this new string
+        Replace "cat:" in the catalogue global attrs by this new string
```
Comment (RondeauG) [inline xscen/utils.py:387]:
```suggestion
        Replace "cat:" in the catalog global attrs by this new string
```

**Review by RondeauG — APPROVED** [review]
Looks good!

File: HISTORY.rst (hunk 1 above)
Comment (aulemahal) [reply HISTORY.rst:5]: Yep.

**Review by juliettelavoie — APPROVED** [review]
ben beau

**Comment by aulemahal** [comment]
@Zeitsperre  J'ai modifié l'action github pour exclure "docs/notebooks".

**Pull request merged**